hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2615,6 +2633,51 @@ func reconcilecontrolPlaneOperatorIngressOperatorRoleBinding(binding *rbacv1.Rol
 	return nil
 }
 
+func reconcileControlPlaneOperatorClusterVersionOperatorClusterRole(role *rbacv1.ClusterRole) error {
+	role.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
```

Comment:
this is non go, why would we need all this perms?

Reply:
I am currently working on utilizing the `tenancy` port of `thanos-querier` (https://cloud.redhat.com/blog/thanos-querier-versus-thanos-querier). That port would only give the hosted CVO metrics for its hosted control plane namespace and only requires the hosted CVO to have following permissions in the hosted control plane namespace: 

```
rules:
  - verbs:
      - get
    apiGroups:
      - metrics.k8s.io/v1beta1
    resources:
      - pods
```

I am also working on removing the need to `get` any `service`. These things needs some work done in the CVO repository.

Reply:
Because the CPO doesn't have any permissions regarding the `metrics.k8s.io/v1beta1` group. Is it better to give the additional permissions to the CPO and then have the CPO give this permission to the CVO? Or have the hypershift-operator give the access to the metrics straight only to the CVO?